dockers/docker-snmp/snmp_yml_to_configdb.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+
+import os
+import yaml
+from swsssdk import ConfigDBConnector
+
+db = ConfigDBConnector()
+db.connect()
+
+snmp_comm_config_db = db.get_table('SNMP_COMMUNITY')
+snmp_config_db_communities = snmp_comm_config_db.keys()
+snmp_general_config_db = db.get_table('SNMP')
+snmp_general_keys = snmp_general_config_db.keys()
+
+full_snmp_comm_list = ['snmp_rocommunity', 'snmp_rocommunities', 'snmp_rwcommunity', 'snmp_rwcommunities']
+
+
+if os.path.exists('/etc/sonic/snmp.yml'):
+    with open('/etc/sonic/snmp.yml', 'r') as yaml_file:
+        yaml_snmp_info = yaml.load(yaml_file)
+
+    for comm_type in full_snmp_comm_list:
+        if comm_type in yaml_snmp_info.keys():
+            if comm_type.startswith('snmp_rocommunities'):
+                for community in yaml_snmp_info[comm_type]:
+                    if community not in snmp_config_db_communities:
+                        db.set_entry('SNMP_COMMUNITY', community, {"TYPE": "RO"})
+            elif comm_type.startswith('snmp_rocommunity'):
+                community = yaml_snmp_info['snmp_rocommunity']
+                if community not in snmp_config_db_communities:
+                    db.set_entry('SNMP_COMMUNITY', community, {"TYPE": "RO"})
+            elif comm_type.startswith('snmp_rwcommunities'):
+                for community in yaml_snmp_info[comm_type]:
+                    if community not in snmp_config_db_communities:
+                        db.set_entry('SNMP_COMMUNITY', community, {"TYPE": "RW"})
+            elif comm_type.startswith('snmp_rwcommunity'):
+                community = yaml_snmp_info['snmp_rwcommunity']
+                if community not in snmp_config_db_communities:
+                    db.set_entry('SNMP_COMMUNITY', community, {"TYPE": "RW"})
+    
+    if yaml_snmp_info['snmp_location']:
+        if 'LOCATION' not in snmp_general_keys:
+            db.set_entry('SNMP', 'LOCATION', {'Location': yaml_snmp_info['snmp_location']})
+    else:
+        pass
+else:
+    pass
```

Comment:
```
else:
    pass
```
This is unnecessary. Suggest logging a message rather than `pass`, so that we have a record of what the script did (or did not do). You can import the `Logger` class from the new `sonic_py_common` library. Here is a simple script which uses it as an example: https://github.com/Azure/sonic-buildimage/blob/master/files/scripts/core_cleanup.py

Also, you can save yourself the extra indenting of the block by changing the condition to:

```python3
if not os.path.exists('/etc/sonic/snmp.yml'):
    <log message here stating that snmp.yml not found, doing nothing>
    sys.exit(1)
```

Note that you will also need to `import sys` for `sys.exit(1)` to work.

Reply:
added.